packages/gestalt/src/PageHeader.js

```diff
@@ -31,48 +89,90 @@ type Props = {|
-            <Heading size="500" lineClamp={1} accessibilityLevel={1}>
-              {title}
-            </Heading>
-            {subtext && (
-              <Box marginTop={2}>
-                <Text lineClamp={1}>{subtext}</Text>
-              </Box>
-            )}
-          </Box>
+  const { text: badgeText, tooltipText: badgeTooltipText } = badge || {};
 
-          {primaryAction && (
-            <Flex alignItems="center" gap={2} flex="none">
-              {secondaryAction}
-              {/* 48px height needed to maintain proper sizing when action is a Link */}
-              <Box alignItems="center" display="flex" marginTop={4} marginBottom={4} height="48px">
-                {primaryAction}
+  return (
+    <div className={borderStyle === 'sm' ? styles.pageHeader : null}>
+      <Box color="white" paddingX={8} paddingY={4} width="100%">
+        <Flex flex="grow" justifyContent="center" maxWidth="100%">
+          <Flex flex="grow" maxWidth={maxWidth}>
+            <Flex.Item minWidth={0} flex="grow" alignSelf="center">
+              <Box marginEnd={6}>
+                <Flex gap={2}>
+                  {thumbnail ? <PageHeaderThumbnail thumbnail={thumbnail} /> : null}
+                  <Flex direction="column" gap={1}>
+                    <Flex alignItems="center">
+                      <PageHeaderTitle hasThumbnail={!!thumbnail} title={title} />
+                      <Box display="none" smDisplay="block" marginStart={badge ? 1 : 3}>
+                        <Flex gap={3}>
+                          {badge ? (
+                            <PageHeaderBadge
+                              badgeText={badgeText}
+                              badgeTooltipText={badgeTooltipText}
+                            />
+                          ) : null}
+                          {helperIconButton ? (
+                            <PageHeaderHelperIconButton
+                              accessibilityLabel={helperIconButton.accessibilityLabel}
+                              accessibilityControls={helperIconButton.accessibilityControls}
+                              accessibilityExpanded={helperIconButton.accessibilityExpanded}
+                              onClick={helperIconButton.onClick}
+                            />
+                          ) : null}
+                        </Flex>
+                      </Box>
+                    </Flex>
+                    {subtext ? (
+                      <PageHeaderSubtext subtext={subtext} helperLink={helperLink} />
+                    ) : null}
+                  </Flex>
+                </Flex>
               </Box>
-            </Flex>
-          )}
+            </Flex.Item>
+            <Flex.Item minWidth={0} flex="none">
+              <Flex gap={4} alignItems={subtext ? undefined : 'center'} height="100%">
+                {items && items.length !== 0 ? <PageHeaderItemsBlock items={items} /> : null}
+                {primaryAction ? (
```

Comment:
Is it a valid assumption that we don't allow a user to pass only a secondary action? What if the page's primary CTA is elsewhere and not in the header?

Reply:
Not allowed based on designs.